File: src/main/java/de/tuberlin/dima/model/StudentInfo.java
package de.tuberlin.dima.model;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class StudentInfo implements Serializable {

    private String name;
    private String major;
    private List<String> courses;

    public StudentInfo(){
        courses = new ArrayList<String>();
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getMajor() {
        return major;
    }

    public void setMajor(String major) {
        this.major = major;
    }

    public List<String> getCourses() {
        return courses;
    }

    public void setCourses(List<String> courses) {
        this.courses = courses;
    }
}

File: src/main/scala/de/tuberlin/dima/model/StudentInfo.scala
package de.tuberlin.dima.model

import scala.collection.mutable.ArrayBuffer

class StudentInfo {

	private var _name: String = _
	private var _major: String = _
	private val _courses: ArrayBuffer[String] = new ArrayBuffer[String]

	def name = _name
	def name(s:String):Unit = {_name = s}

	def major = _major
	def major(s:String):Unit = {_major = s}

	def courses = _courses
	def courses(arrayBuffer: ArrayBuffer[String]): Unit = {
		_courses ++=(arrayBuffer)
	}

	override def toString: String = {
		_name + " " + _major + " [" + _courses.foreach(print(_)+";") + "]"
	}

}